base/array.jl

```diff
@@ -212,6 +212,37 @@ function getindex(::Type{Any}, vals::ANY...)
 end
 getindex(::Type{Any}) = Array{Any,1}(0)
 
+"""
+    fill!(A, x)
+
+Fill array `A` with the value `x`. If `x` is an object reference, all elements will refer to
+the same object. `fill!(A, Foo())` will return `A` filled with the result of evaluating
+`Foo()` once.
+
+```jldoctest
```

Comment:
While you are at it, perhaps add an `# Examples` heading before the doctests (as in #22522)? :)